### ISSUE TITLE
How to handle Time Out

### ISSUE THREAD

**Issue by rodrigoavalente** [fluttercommunity/flutter_webview_plugin#444]
Hello, I'm trying to do something with the webview if a time out error occurs, however, the event is not triggered when the time out happens on my device.

```
StreamSubscription _onDestroy;
StreamSubscription<String> _onUrlChanged;
StreamSubscription<WebViewStateChanged> _onStateChanged;  StreamSubscription<WebViewHttpError> _onHttpError;

...
@override
void initState() {
    super.initState();

    ...
   _onHttpError = _flutterWebviewPlugin.onHttpError.listen((WebViewHttpError error) {
       // TODO: implement
       print(error.code);
   });
}
```

**Comment by scofield-hello** [comment]
I have same problem.

**Comment by bhjSky** [comment]
I have the same problem.

**Comment by scofield-hello** [comment]
@rodrigoavalente did you solve the problem yet?

**Comment by rodrigoavalente** [comment]
@scofield-hello unfortunally I don't. I'm trying to use some gimmicks to check if connection is available and send the user to another view with some message, but it is not perfect.

**Comment by rodrigoavalente** [comment]
Hello, sorry for the late reply, I had to stop working on the app that was using this plugin, and I just had returned to working on it, from my investigations I found that the `Connection Refused` and `Connection Timeout`  errors are returning `-6` and `-8` on android and as the class `WebViewHttpError` expects two strings as parameters an exception is thrown and nothing happens, so the event is not fired.

```
// base.dart file function that handles the events
 Future<Null> _handleMessages(MethodCall call) async {
    switch (call.method) {
      case 'onBack':
        _onBack.add(null);
        break;
      case 'onDestroy':
        _onDestroy.add(null);
        break;
      case 'onUrlChanged':
        _onUrlChanged.add(call.arguments['url']);
        break;
      case 'onScrollXChanged':
        _onScrollXChanged.add(call.arguments['xDirection']);
        break;
      case 'onScrollYChanged':
        _onScrollYChanged.add(call.arguments['yDirection']);
        break;
      case "onProgressChanged":
        _onProgressChanged.add(call.arguments["progress"]);
        break;
      case 'onState':
        _onStateChanged.add(
          WebViewStateChanged.fromMap(
            Map<String, dynamic>.from(call.arguments),
          ),
        );
        break;
      case 'onHttpError':
        _onHttpError.add(WebViewHttpError(call.arguments['code'], call.arguments['url']));
        break;
    }
  }

// base.dart file WebViewHttpError
class WebViewHttpError {
  WebViewHttpError(this.code, this.url);

  final String url;
  final String code;
}
```

Well, I don't know if the returning codes are the expected on android, I would appreciate if someone could provide some advice on this, thanks for the help.

**Comment by rodrigoavalente** [comment]
I'll close this, I've noticed that problem was fixed on 0.3.6.
